src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -117,10 +117,31 @@ private Task<bool> NominateProjectAsync(string projectUniqueName, IVsProjectRest
                     $"The nominate API is called for '{projectUniqueName}'.");
 
                 var projectNames = ProjectNames.FromFullProjectPath(projectUniqueName);
+                DependencyGraphSpec dgSpec;
+                IReadOnlyList<IAssetsLogMessage> nominationErrors = null;
+                try
+                {
+                    dgSpec = ToDependencyGraphSpec(projectNames, projectRestoreInfo, projectRestoreInfo2);
+                }
+                catch (Exception e)
+                {
+                    var restoreLogMessage = RestoreLogMessage.CreateError(NuGetLogCode.NU1010, string.Format(Resources.NU1010, e.Message));
+                    restoreLogMessage.ProjectPath = projectUniqueName;
+
+                    nominationErrors = new List<IAssetsLogMessage>()
+                    {
+                        AssetsLogMessage.Create(restoreLogMessage)
+                    };
+
+                    var projectDirectory = Path.GetDirectoryName(projectUniqueName);
+                    string projectIntermediatePath = projectRestoreInfo == null
+                        ? projectRestoreInfo2.BaseIntermediatePath
+                        : projectRestoreInfo.BaseIntermediatePath;
+                    var dgSpecOutputPath = GetProjectOutputPath(projectDirectory, projectIntermediatePath);
+                    dgSpec = DependencyGraphSpec.CreateMinimal(projectUniqueName, dgSpecOutputPath);
```

Comment:
Will be the nomination errors added to the assets file?

Reply:
yes, the exception is added to the message list on lines 131 to 134, and on line 144 it gets added to the project system cache, which is then retrieved in NuGet.Commands's RestoreCommand.